src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -947,18 +946,27 @@ void MetaspaceShared::initialize_runtime_shared_and_meta_spaces() {
     // The base archive cannot be mapped. We cannot dump the dynamic shared archive.
     AutoCreateSharedArchive = false;
     DynamicDumpSharedSpaces = false;
-    FileMapInfo::fail_continue("Unable to map shared spaces");
+    log_info(cds)("Unable to map shared spaces");
     if (PrintSharedArchiveAndExit) {
       vm_exit_during_initialization("Unable to use shared archive.");
+    } else if (RequireSharedSpaces) {
+      FileMapInfo::fail_stop("Unable to map shared spaces");
```

Comment:
Looks odd to use `fail_stop` here and `vm_exit_during_initialization` just before it. ??